packages/web-pkg/src/components/AppTemplates/AppWrapper.vue

```diff
@@ -466,6 +466,22 @@ export default defineComponent({
       }
     })
 
+    const originalDownloadFileActionHandler = unref(downloadFileActions)[0].handler
+
+    /**
+     * The interceptor is used to save the file automatically when in dirty state,
+     * so the downloaded file equals the current state
+     */
+    const downloadFileActionInterceptor = async (args: FileActionOptions) => {
+      if (unref(isDirty)) {
+        await save()
+        autosavePopup()
+      }
+      originalDownloadFileActionHandler(args)
+    }
+
+    unref(downloadFileActions)[0].handler = downloadFileActionInterceptor
```

Comment:
Overwriting the handler of the action feels a bit hacky to me 🤔 Can't you just add a method that checks for the dirty state and then calls `unref(downloadFileActions)[0].handler`?

Reply:
I don't get it, we need to save on download, how to "just add a method" ?

Reply:
I thought the component calls `unref(downloadFileActions)[0].handler` directly, but that doesn't seem to be the case. Hmm. Then I don't have a better idea...

Reply:
It's not suuuper nice, but I don't think it's too bad. Fine for me.

The only issue I have with this, is that you `unref` in the setup function - the current code will break when the computed in the composable is reevaluated and there's a new action object.
This needs to live inside a watcher to make it more resilient - I'm sorry 😅